### PR TITLE
bugfix/COR-1865-no-auto-resolve-1970

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard29**
Throw error if pageMetric not found in data

### CODE REVIEW

File: packages/app/src/utils/get-last-insertion-date-of-page.ts [hunk 1]
```diff
@@ -6,17 +6,15 @@ function hasLastValue(metric: any): boolean {
 }
 
 function hasValues(metric: any): boolean {
-  return Array.isArray(metric?.values) &&
-    typeof metric?.values[metric.values.length - 1]?.date_of_insertion_unix !== 'undefined';
+  return Array.isArray(metric?.values) && typeof metric?.values[metric.values.length - 1]?.date_of_insertion_unix !== 'undefined';
```
Comment (yorickdevries) [inline packages/app/src/utils/get-last-insertion-date-of-page.ts:9]: Nothing is changed except whitespace, consider removing the change

File: packages/app/src/utils/get-last-insertion-date-of-page.ts [hunk 2]
```diff
@@ -6,17 +6,15 @@ function hasLastValue(metric: any): boolean {
...
 }
 
 function hasInsertionDate(metric: any): boolean {
   return typeof metric?.date_of_insertion_unix !== 'undefined';
 }
 
 function hasNestedLastValue(metric: any): boolean {
-  return Array.isArray(metric?.values) &&
-    typeof metric?.values[0]?.last_value?.date_of_insertion_unix !== 'undefined';
+  return Array.isArray(metric?.values) && typeof metric?.values[0]?.last_value?.date_of_insertion_unix !== 'undefined';
```
Comment (yorickdevries) [inline packages/app/src/utils/get-last-insertion-date-of-page.ts:17]: Nothing is changed except whitespace, consider removing the change

File: packages/app/src/utils/get-last-insertion-date-of-page.ts [hunk 3]
```diff
@@ -33,7 +31,7 @@ function getDateFromInsertionDate(metric: any): number {
 }
 
 function getDateFromNestedLastValue(metric: any): number {
-  return metric?.values.reduce((lastDate :number, innerValue: any) => {
+  return metric?.values.reduce((lastDate: number, innerValue: any) => {
```
Comment (yorickdevries) [inline packages/app/src/utils/get-last-insertion-date-of-page.ts:34]: Nothing is changed except whitespace, consider removing the change

File: packages/app/src/utils/get-last-insertion-date-of-page.ts [hunk 4]
```diff
@@ -55,14 +53,21 @@ function getMetricDate(metricOrUnixDate: any): number {
   return 0;
 }
 
-export function getLastInsertionDateOfPage(
-  data: unknown,
-  pageMetrics: string[]
-) {
-  return pageMetrics.reduce((lastDate, metricProperty) => {
+export function getLastInsertionDateOfPage(data: unknown, pageMetrics: string[]) {
+  const foundMetrics: string[] = pageMetrics.filter((metricProperty) => {
```
Comment (yorickdevries) [inline packages/app/src/utils/get-last-insertion-date-of-page.ts:57]: Naming foundMetrics is not really clear; consider renaming to definedMetrics or merticsWithValues

File: packages/app/src/utils/get-last-insertion-date-of-page.ts [hunk 5]
```diff
@@ -55,14 +53,21 @@ function getMetricDate(metricOrUnixDate: any): number {
...
+    if (get(data, metricProperty) !== undefined) {
```
Comment (yorickdevries) [inline packages/app/src/utils/get-last-insertion-date-of-page.ts:58]: The filter method should return a boolean. And there is no else case.
Consider: ```return typeof get(data, metricProperty) === 'number';```

File: packages/app/src/utils/get-last-insertion-date-of-page.ts [hunk 6]
```diff
@@ -55,14 +53,21 @@ function getMetricDate(metricOrUnixDate: any): number {
...
+      return metricProperty;
+    }
+  });
+
+  if (foundMetrics?.length === 0) {
```
Comment (yorickdevries) [inline packages/app/src/utils/get-last-insertion-date-of-page.ts:63]: why the ```?``` in ```foundMetrics?```. Seems redundant and even error prone.

File: packages/app/src/utils/get-last-insertion-date-of-page.ts [hunk 7]
```diff
@@ -55,14 +53,21 @@ function getMetricDate(metricOrUnixDate: any): number {
...
+    throw new Error(`Pagemetrics not found in data`);
+  }
+
+  return foundMetrics.reduce((lastDate, metricProperty) => {
     const metric: any = get(data, metricProperty);
     const metricDate = getMetricDate(metric);
-    
+
```
Comment (yorickdevries) [inline packages/app/src/utils/get-last-insertion-date-of-page.ts:70]: why is this new line added?

File: packages/app/src/utils/get-last-insertion-date-of-page.ts [hunk 8]
```diff
@@ -55,14 +53,21 @@ function getMetricDate(metricOrUnixDate: any): number {
...
     return Math.max(metricDate, lastDate);
   }, 0);
-};
+}
```
Comment (yorickdevries) [inline packages/app/src/utils/get-last-insertion-date-of-page.ts:73]: why is the semi-colon removed?

**Review by yorickdevries — CHANGES_REQUESTED** [review]
Please consider the comments in the review and take a look at the tests when resolved. Perhaps you can even add a test which checks for the behavior when Pagemetrics is missing.

File: packages/app/src/utils/get-last-insertion-date-of-page.ts (hunk 1 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/utils/get-last-insertion-date-of-page.ts]: Reformat done automatically by Husky commit hook. Forceably undone now.

File: packages/app/src/utils/get-last-insertion-date-of-page.ts (hunk 2 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/utils/get-last-insertion-date-of-page.ts]: Reformat done automatically by Husky commit hook. Forceably undone now.

File: packages/app/src/utils/get-last-insertion-date-of-page.ts (hunk 3 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/utils/get-last-insertion-date-of-page.ts]: Reformat done automatically by Husky commit hook. Forceably undone now.

File: packages/app/src/utils/get-last-insertion-date-of-page.ts (hunk 7 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/utils/get-last-insertion-date-of-page.ts]: Reformat done automatically by Husky commit hook. Forceably undone now.

File: packages/app/src/utils/get-last-insertion-date-of-page.ts (hunk 8 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/utils/get-last-insertion-date-of-page.ts]: Reformat done automatically by Husky commit hook. Forceably undone now.

File: packages/app/src/utils/get-last-insertion-date-of-page.ts (hunk 4 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/utils/get-last-insertion-date-of-page.ts]: Renamed to metricsAvailableInData

File: packages/app/src/utils/get-last-insertion-date-of-page.ts (hunk 5 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/utils/get-last-insertion-date-of-page.ts]: Implemented suggestion.

File: packages/app/src/utils/get-last-insertion-date-of-page.ts (hunk 6 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/utils/get-last-insertion-date-of-page.ts]: Removed optional

**Comment by yorickdevries** [comment]
Thanks for the changes. The PR looks much cleaner now. If Husky changes so much formatting, perhaps we should straighten th whole codebase so everything is linted correctly. Furthermore, a test is still failing. Can you look into that?

File: packages/app/src/utils/get-last-insertion-date-of-page.ts (hunk 8 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/utils/get-last-insertion-date-of-page.ts]: Newline added

**Review by yorickdevries — APPROVED** [review]
approved

**Pull request merged**